examples/payjoin-0.13.0.md

```diff
@@ -0,0 +1,551 @@
+# Introduction
+
+This is a tutorial to get you started with payjoin. We will
+demonstrate how you can setup both sender and receiver sides
+in a payjoin transaction.
+
+We do a simple bitcoin wallet setup with `bitcoincore_rpc`,
+for that part, in production, you are expected to use
+your own wallet implementation.
+
+The tutorial demosntrates version 1 of payjoin as described
+in BiP0078.
+
+For any feedback, please free to reach out through the payjoin
+official discord or github.
+	
+# 1. Project Setup
+
+  First lets setup a new rust project and call it `payjoin-tutorial`
+  ```shell
+	cargo new payjoin-tutorial
+  ```
+
+  Now we should have a rust project with a src folder a `cargo.toml`
+  file.
+  
+  
+  Add required dependecies to `cargo.toml` so it looks like the
+  following:
+
+  ```toml
+  [dependencies]
+  axum = "0.7.2" 
+  bitcoincore-rpc = "0.17.0" 
+  http-body-util = "0.1.0"
+  payjoin = { version = "0.13.0", features = ["send", "receive", "base64"] }
+  tokio = {version = "1.35.1", features = ["rt", "macros", "rt-multi-thread"] }
+  ureq = "2.9.1" 
+  ```
+
+	
+# 2. Wallets
+
+  We will setup basic bitcoin wallet utilising bitcoin core in Regtest
+  env.
+
+  Please refer to Bitcoin Core to setup Regtest localy.
+
+  Once you setup Bitcoin Core, create two wallets from the cli:
+
+  ```shell
+   ./src/bitcoin-cli createwallet sender
+  ```
+
+  ```shell
+   ./src/bitcoin-cli createwallet receiver
+  ```
+
+  Make sure your wallets are loaded
+
+  ```shell
+   ./src/bitcoin-cli loadwallet sender
+  ```
+
+  ```shell
+   ./src/bitcoin-cli loadwallet receiver
+  ```
+
+   Now lets get some bitcoins in tothe wallets.
+
+   First lets get new address for the sender:
+
+   ```shell
+    ./src/bitcoin-cli   -rpcwallet=sender getnewaddress
+   ```
+
+   Get bitcoins for the sender:
+   ```shell
+     ./src/bitcoin-cli   -rpcwallet=receiver generatetoaddress 250 <sender_address>
+   ```
+
+   And the same process for the receiver:
+
+   ```shell
+    ./src/bitcoin-cli   -rpcwallet=receiver getnewaddress
+   ```
+   ```shell
+     ./src/bitcoin-cli   -rpcwallet=receiver generatetoaddress 250 <receiver_address>
+   ```
+
+
+  Ok now we setup the RPC client to talke to the above wallets we just
+  created.
+  ```shell
+  touch payjoin-tutorial/src/wallet.rs
+  ```
```

Comment:
Examples should be self contained and able to be re-run without side effects. That's what differentiates them from e.g. payjoin-cli. Can we implement this setup using rpc automation as in integration tests or otherwise an in-memory BDK database?

Reply:
why does this need to run in a web environment?

Reply:
Alternatively, you could scrap axum for blocking, simple rouille and just put payjoin-cli on its own thread